src/AutoRest.CSharp/Common/Input/Configuration.cs

```diff
@@ -600,7 +603,10 @@ private static void WriteConfiguration(Utf8JsonWriter writer)
             {
                 MgmtTestConfiguration.SaveConfiguration(writer);
             }
-            WriteIfNotDefault(writer, Options.Branded, ApiTypes is AzureApiTypes);
+            if (ApiTypes is AzureApiTypes)
+            {
+                writer.WriteString(Options.Flavor, "azure");
+            }
```

Comment:
I think we could remove this check.
When it is `AzureApiTypes`, the value is azure, and when it is not, we should output whatever it is (and omit it if it is null).

Reply:
change to save the `flavor` value if possible